pkg/apiserver/webhooks.go

```diff
@@ -82,9 +83,48 @@ func NewWebhookConfig() *WebhookConfig {
 		!hasEnv("WEBHOOK_HOST") {
 
 		cfg.Type = WebhookTypeService
-		cfg.ServiceName = getEnv("WEBHOOK_SERVICE_NAME", "api")
-		cfg.ServiceNamespace = getEnv("WEBHOOK_SERVICE_NAMESPACE", "porch-system")
-		cfg.ServiceNamespace = getEnv("CERT_NAMESPACE", cfg.ServiceNamespace)
+
+		var apiSvcNs string
+		cfg.ServiceName = os.Getenv("WEBHOOK_SERVICE_NAME")
+		if cfg.ServiceName == "" { // empty value and unset envvar are the same for our purposes
+			// if WEBHOOK_SERVICE_NAME is not set, try to use the porch API service name
+			apiSvc, err := util.GetPorchApiServiceKey(ctx)
+			if err != nil {
+				panic(fmt.Sprintf("WEBHOOK_SERVICE_NAME environment variable is not set, and could not find porch APIservice: %v", err))
```

Comment:
Should we try to recover instead of panic? Can we run without the webhook?

Reply:
Without a working webhook nothing prevents actually deleting PackageRevisions that are not in the DeletionProposed state. As such the API server cannot properly run without the webhook, so IMHO it is better to panic here,  than trying to recover.